matlab/test/arrow/args/tParseValidElements.m

```diff
@@ -0,0 +1,163 @@
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+
+classdef tParseValidElements < matlab.unittest.TestCase
+    % Tests for arrow.args.parseValidElements
+    
+    methods(Test)
+        % Test methods
+        function InferNullsTrue(testCase)
+        % Verify values for which ismissing returns true for are not
+        % considered valid.
+            data = [1 NaN 3 NaN 5];
+            validElements = parseValidElements(data, InferNulls=true);
+            expectedValidElements = [true; false; true; false; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+
+            data = [1 2 3];
+            validElements = parseValidElements(data, InferNulls=true);
+            expectedValidElements = [true; true; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+        end
+
+        function InferNullsFalse(testCase)
+        % Verify all elements are treated as Valid wen InferNulls=false is
+        % provided - including values for which that ismissing returns true.
+            data = [1 NaN 3 NaN 5];
+            validElements = parseValidElements(data, InferNulls=false);
+            expectedValidElements = [true; true; true; true; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+        end
+
+        function LogicalValid(testCase)
+            data = [1 2 3]; 
+
+            % Supply a scalar true value for Valid 
+            validElements = parseValidElements(data, Valid=true);
+            expectedValidElements = [true; true; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+
+            % Supply a scalar false value for Valid
+            validElements = parseValidElements(data, Valid=false);
+            expectedValidElements = [false; false; false];
+            testCase.verifyEqual(validElements, expectedValidElements);
+
+            % Supply a logical vector for Valid
+            validElements = parseValidElements(data, Valid=[false; true; true]);
+            expectedValidElements = [false; true; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+        end
+
+        function NumericValid(testCase)
+            data = [1 2 3]; 
+
+            % Supply 1 valid index for Valid
+            validElements = parseValidElements(data, Valid=2);
+            expectedValidElements = [false; true; false];
+            testCase.verifyEqual(validElements, expectedValidElements);
+
+            % Supply a numeric vector for Valid 
+            validElements = parseValidElements(data, Valid=[1 3]);
+            expectedValidElements = [true; false; true];
+            testCase.verifyEqual(validElements, expectedValidElements);
+        end
+
+        function ComplexValidIndicesError(testCase)
+        % Verify parseValidElements errors if Valid is a numeric array
+        % containing complex numbers.
```

Comment:
```suggestion
            % Verify parseValidElements errors if Valid is a numeric array
            % containing complex numbers.
```

Is this natural comment indent style in MATLAB...?

Reply:
Either comment style works well. Typically, comments that describe the overall behavior of a function are not indented. These comments become the help text that appears when you run: 

```MATLAB
>> help <functionName>
```
Here's a [link](https://www.mathworks.com/help/matlab/matlab_prog/add-help-for-your-program.html) to a documentation page about help text if you are interested in reading more about it. However, it's fine for these comments to be indented, so it's really just a personal preference.

Reply:
> Typically, comments that describe the overall behavior of a function are not indented.

Oh, I see. Then could you revert my suggestions? Sorry.

Reply:
I think it's ok if we leave the comments indented. Both styles are acceptable in MATLAB. I talked with @kevingurney about this yesterday and he prefers indenting these comments. I don't have a strong opinion, so I'm happy to go with indenting them. Does that work for you @kou?

Reply:
OK!
I'll merge this after you push the changes.

Reply:
Thanks Kou for working with us!